### DISCUSSION TITLE
How I can use LLM transformer to apply LLM for all data source output and shcema type?

### DISCUSSION THREAD

**Discussion by YuriyGavrilov** [apache/seatunnel#7565] [Q&A] [answered]
Just in case: I have one table with some 10 fields or it will be all tables in data source like for while each table in datasource.
Next step it will be LLM to identify sensitive data in the source output. field names and 10 lines of real data limit should be taken in considiration. The output of LLM should be a json string with field names, identified data category and flag true or false about sensitive. So what do you think is it possible? Can't imagine how to collect all the table schema and rows and send to LLM transformer in one request.

**Comment by Hisoka-X** [comment, marked as answer]
I haven't actually tried it, but it's theoretically possible. For each row of data, we convert it to json and send it to LLM.
eg:
| name | age | loc |
| --- | --- | --- |
| Jack | 24 | USA |
| Tom | 18 | ENG |

```
{"name":"Jack", "age": 24, "loc": "USA"}      --> LLM (prompt: "check...., the output of LLM should be a json string with field names, identified data category and flag true or false about sensitive") 
{"name":"Tom", "age": 18, "loc": "ENG"}    --> LLM (prompt: "check...., the output of LLM should be a json string with field names, identified data category and flag true or false about sensitive") 
```

The output by LLM transform:
| name | age | loc | llm_output |
| --- | --- | --- | --- |
| Jack | 24 | USA | {"name":false, "age": true, "loc": true}  |
| Tom | 18 | ENG | {"name":false, "age": true, "loc": true} |

Then you can use [JsonPath](https://seatunnel.apache.org/docs/2.3.7/transform-v2/jsonpath) transform to process the json data.
